api/resource.rb

```diff
@@ -78,6 +78,8 @@ module Properties
       attr_reader :update_verb
       # [Optional] The HTTP verb used during delete. Defaults to :DELETE.
       attr_reader :delete_verb
+      # [Optional] An array of function names that determine whether an error is retryable.
+      attr_reader :retry_predicates
```

Comment:
Wdyt of naming this something like `custom_error_retryability_checks`? `retry_predicates` is pretty general, and it's a little hard to tell at a glance what it means. I think we want to convey with the naming that:

* these are checks if an error is retryable

* they're extra, in addition to default checks that already exist

Reply:
I think the fact that they're a `resource_override` in `terraform.yaml` is sufficient to infer that the checks are in addition to any standard retries. I plan to refactor the resource specific logic from `isRetryableError` into additional checks defined in `third_party/terraform/utils/error_retry_predicates.go` in a future PR. `predicate` unmistakably specifies the return type of the functions.

Reply:
Fair enough!